.github/workflows/image-signer.yaml

```diff
@@ -0,0 +1,69 @@
+name: Pull and Sign Container Image
+
+on:
+  repository_dispatch:
+    types: [sign-image]  
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  id-token: write # Required for OIDC token access
+
+jobs:
+  build-and-sign:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the repository
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Log in to Docker Hub
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          registry: docker.io
+          username: ${{ github.event.client_payload.docker_username }}
```

Comment:
@uniqueg do you think Container Registry will remain same for one pubgrade instance? If yes, then we can move these registry credentials to Github Secrets.

Reply:
_Instance_: yes. _Implementation_: no. That is, one organization deploying Pubgrade will almost certainly use a different set of credentials than another organization deploying their own Pubgrade instance.

Would one code repository use different instances/deployments of Pubgrade, connected with different registries? Possibly. But it's not a feature we absolutely have to support. At least not now.

Does that answer your question?